src/spaceObjects/spaceshipParts/weaponTube.cpp

```diff
@@ -306,63 +306,108 @@ string WeaponTube::getTubeName()
-        auto turn_exit = turn_center + vec2FromAngle(missile_angle - left_or_right) * turn_radius;
-        if (target_velocity_length < 1.0f)
+        missile_angle = std::numeric_limits<float>::infinity();
+    }
+    else
+    {
+        const float tube_angle = glm::radians(parent->getRotation() + direction);
+        const float turn_rate = glm::radians(missile.turnrate);
+        const float turn_radius = missile.speed / turn_rate;
+
+        // Get target parameters in the tube centered reference frame:
+        const glm::vec2 target_position = transform_global_to_tube(target->getPosition() - parent->getPosition(), tube_angle);
+        const glm::vec2 target_velocity = transform_global_to_tube(target->getVelocity(), tube_angle);
+
+        const int MAX_ITER = 10;
+        const float tolerance = 0.1f * target->getRadius();
+        bool converged = false;
+        glm::vec2 aim_position = target_position; // Set initial aim point
+        float turn_direction; // Left: -1, Right: +1, No turn: 0
+        float turn_angle; // In radians. Value of 0 means no turn.
+        for (int iterations=0; iterations<MAX_ITER && converged == false; iterations++)
         {
-            //If the target is almost standing still, just target the position directly instead of using the velocity of the target in the calculations.
-            float time_missile = glm::length(turn_exit - target_position) / data.speed;
-            auto interception = turn_exit + vec2FromAngle(missile_angle) * data.speed * time_missile;
-            float r = target->getRadius() / 2;
-            if (glm::length2(interception - target_position) < r*r)
-                return missile_angle;
-            missile_angle = vec2ToAngle(target_position - turn_exit);
+            // Select turn direction
+            // Turn in the direction of the target on condition that the target
+            // is not inside the turning circle of that side. If it is inside
+            // the turning circle, turn in the opposite direction.
+            const float r_left = glm::length(aim_position - glm::vec2(-turn_radius, 0.0f));
+            const float r_right = glm::length(aim_position - glm::vec2(turn_radius, 0.0f));
+            if ((aim_position.x < 0.0f && r_left >= turn_radius) || r_right < turn_radius)
+                turn_direction = -1.0f;
+            else if ((aim_position.x >= 0.0f && r_right >= turn_radius) || r_left < turn_radius)
+                turn_direction = 1.0f;
+            else
+                turn_direction = 0.0f;
+            
+            // Calculate turn angle
+            if (turn_direction == 0.0f)
+            {
+                turn_angle = 0.0f;
+            }
+            else
+            {
+                const float x = turn_direction * aim_position.x;
+                const float aim_radius = glm::length(aim_position - turn_direction*glm::vec2(turn_radius, 0.0f));
+                const float a = glm::acos((turn_radius - x) / aim_radius);
```

Comment:
Instead of acos could atan2 be used? Or https://github.com/daid/SeriousProton/blob/master/src/vectorUtils.h#L15C26-L15C26
acos has a pretty high risk of a domain error causing NaN

Reply:
Good point. I will update the code to try this.